frontend/src/ui/Blocks/Series.tsx

```diff
@@ -193,46 +218,412 @@ type SeriesBlockContainerProps = {
+                ":not(:last-child)": {
+                    borderBottom: "1px solid var(--grey86)",
+                },
+                ":last-child button": {
+                    borderRadius: "0 0 8px 8px",
+                },
+            },
+        },
+    };
+
+    const handleBlur = (event: React.FocusEvent<HTMLUListElement, Element>) => {
+        if (!event.currentTarget.contains(event.relatedTarget as HTMLUListElement)) {
+            close();
+        }
+    };
+
+    const list = match(type, {
+        view: () => <>
+            <div>{t("series.settings.view")}</div>
+            <ul role="menu" onBlur={event => handleBlur(event)}>
+                <MenuItem
+                    disabled={viewState === "slider"}
+                    onClick={() => setViewState("slider")}
+                    close={() => close()}
+                    Icon={FiColumns}
+                    label={t("series.settings.slider")}
+                />
+                <MenuItem
+                    disabled={viewState === "gallery"}
+                    onClick={() => setViewState("gallery")}
+                    close={() => close()}
+                    Icon={FiGrid}
+                    label={t("series.settings.gallery")}
+                />
+                <MenuItem
+                    disabled={viewState === "list"}
+                    onClick={() => setViewState("list")}
+                    close={() => close()}
+                    Icon={FiList}
+                    label= {t("series.settings.list")}
+                />
+            </ul>
+        </>,
+        order: () => <>
+            <div>{t("series.settings.order")}</div>
+            <ul role="menu" onBlur={event => handleBlur(event)}>
+                <MenuItem
+                    disabled={eventOrder === "NEW_TO_OLD"}
+                    onClick={() => setEventOrder("NEW_TO_OLD")}
+                    close={() => close()}
+                    label= {t("series.settings.new-to-old")}
+                />
+                <MenuItem
+                    disabled={eventOrder === "OLD_TO_NEW"}
+                    onClick={() => setEventOrder("OLD_TO_NEW")}
+                    close={() => close()}
+                    label= {t("series.settings.old-to-new")}
+                />
+            </ul>
+        </>,
+    });
+
+    return (
+        <Floating hideArrowTip borderWidth={0} css={listStyle}>
+            {list}
+        </Floating>
+    );
+};
+
+type MenuItemProps = {
+    Icon?: IconType;
+    label: string;
+    onClick: () => void;
+    close: () => void;
+    disabled?: boolean;
+};
+
+const MenuItem: React.FC<MenuItemProps> = ({ Icon, label, onClick, close, disabled }) => {
+    const ref = useRef<HTMLButtonElement>(null);
+    const preventEnter = (event: React.KeyboardEvent<HTMLButtonElement>) => {
+        if (ref.current?.getAttribute("aria-disabled") === "true"
+                && event.key === "Enter") {
+            event.preventDefault();
+        }
+    };
```

Comment:
Then we could also just use `disabled` on the buttons instead of `aria-disabled`, right?
The reasoning behind this (admittedly not too pretty) function was to specifically keep the ability to focus the disabled buttons as I was under the impression that this would be the right approach for accessibility concerns. But now that I'm trying to find a good reference, I struggle to do so and realize that I might have misunderstood or misremembered some things about this.

Reply:
Ah interesting. Yeah I think my intuition would be to not make it tab-selectable if it's disabled. But I also have no source on whether that's the right thing to do. Maybe let's go with the simpler option for now.